### PR TITLE
Mockup for the Notification's group filters

### PR DESCRIPTION
**Pull request by danidoni**
Screenshot from 2021-02-01 15-57-31

## TODO
- [ ] Fix code climate issues

_Replace this line with a description of your changes. Please follow the suggestions in the comment below._

-----------------

If this PR requires any particular action or consideration before deployment,
please check the reasons or add your own to the list:

* [ ] Requires a database migration that can cause downtime. Postpone deployment until maintenance window[1].
* [ ] Contains a data migration that can cause temporary inconsistency, so should be run at a specific point of time.
* [ ] Changes some configuration files (e.g. options.yml), so the changes have to be applied manually in the reference server.
* [ ] A new Feature Toggle[2] should be enabled in the reference server.
* [ ] Proper documentation or announcement has to be published upfront since the introduced changes can confuse the users.

[1] https://github.com/openSUSE/open-build-service/wiki/Deployment-of-build.opensuse.org#when-there-are-migrations
[2] https://github.com/openSUSE/open-build-service/wiki/Feature-Toggles-%28Flipper%29#you-want-real-people-to-test-your-feature

### CODE REVIEW

**Comment by danidoni** [comment]
[Filter by group](https://trello.com/c/d9PDB2mD/1604-filter-by-group)

File: src/api/app/controllers/webui/users/notifications_controller.rb [hunk 1]
```diff
@@ -91,9 +82,9 @@ def paginated_notifications
   end
 
   def notifications_filter
-    NotificationsFilterPresenter.new(projects_for_filter,
+    NotificationsFilterPresenter.new(ProjectFilterChoicesPresenter.new(params[:project]),
+                                     GroupFilterChoicesPresenter.new,
```
Comment (danidoni) [inline src/api/app/controllers/webui/users/notifications_controller.rb:86]: What do you think of this kind of refactor? The thing is, with the group filtering now there are 5 parameters. Rubocop complains about that. I've been thinking in disabling that cop here but first I want to try this approach.

**Comment by krauselukas** [comment]
@danidoni the mockup it self looks good to me, I would have only one thing in mind which might improve the handling a bit in case you are member of many groups. Maybe a way to collapse the filter group area. Otherwise you might need to scroll a bit to reach the project filters.

File: src/api/app/controllers/webui/users/notifications_controller.rb (hunk 1 above)
Comment (vpereira) [reply src/api/app/controllers/webui/users/notifications_controller.rb:86]: Better if we could switch from positional argument to keywords. In another hand if you are depending exclusively on params and coupling the classes together isn't a problem, then you could pass params to the `NotificationsFilterPresenter` and in the initializer you create the dependency with the `ProjectFilterChoicesPresenter` and `GroupFilterChoicesPresenter`

**Comment by saraycp** [comment]
This was just a Mockup. We can close the PR.

**Pull request closed without merging**